app/services/group_service.rb

```diff
@@ -41,6 +41,7 @@ def self.update(group:, params:, actor:)
 
   def self.archive(group:, actor:)
     actor.ability.authorize! :archive, group
+    SubscriptionService.new(group, actor).end_subscription!
```

Comment:
Looks like it bypasses the perm check if you don't pass in an actor to `SubscriptionService#new`

```
def initialize(group, actor = nil)
  actor.ability.authorize!(:choose_subscription_plan, group) if actor
```